tileview/src/main/java/com/qozix/tileview/tiles/TileCanvasViewGroup.java

```diff
@@ -31,6 +31,7 @@
   private TileRenderTask mLastRunTileRenderTask;
 
   private DetailLevel mDetailLevelToRender;
+  private DetailLevel mLastRequestedDetailLevel;
```

Comment:
great explanation, and good point about the possibility of having to render a ton of tiles if scaled down to a point where all tiles are visible.  it'd be even worse if the user had only a single, very large detail level, and forced all those tiles to be rendered at once - it'd definitely OOM (that'd fail, silently, since decode OOM are caught, but still - there'd be missing pieces).  that said, i think that it's the user's responsibility if he allows that to happen - we can make reasonable attempts to be as responsible as possible, but we can't avoid irresponsible use-cases.

anyway, i'm fine with the change here (adding `mLastRequestedDetailLevel`) - it sounds like a reasonable way to handle the edge case you're describing.